packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -1339,9 +1339,82 @@ abstract class RenderViewportBase2<ParentDataClass extends ContainerParentDataMi
   }
 
   @override
-  double getOffsetToReveal(RenderObject descendant, double alignment) {
-    // TODO(abath): Implement this function for sliver-based viewports.
-    return 0.0;
+  double getOffsetToReveal(RenderObject target, double alignment) {
+    double leadingScrollOffset;
+    double targetMainAxisExtent;
+    RenderObject descendant;
+
+    if (target is RenderBox) {
+      final RenderBox targetBox = target;
+
+      RenderBox pivot = targetBox;
+      while (pivot.parent is RenderBox)
+        pivot = pivot.parent;
+
+      assert(pivot.parent != null);
+      assert(pivot.parent != this);
+      assert(pivot != this);
+
+      final Matrix4 transform = targetBox.getTransformTo(pivot);
+      final Rect bounds = MatrixUtils.transformRect(transform, targetBox.paintBounds);
+
+      target = pivot;
+      assert(target.parent is RenderSliver);
```

Comment:
maybe add TODO here mentioning that we should support other coordinate systems more generically